src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingGlassblower.java

```diff
@@ -147,6 +127,43 @@ public boolean canRecipeBeAdded(final IToken<?> token)
         return false;
     }
 
+    @Override
+    public void onColonyTick(@NotNull final IColony colony)
+    {
+        super.onColonyTick(colony);
+    }
+
+    @Override
+    public void checkForWorkerSpecificRecipes()
+    {
+        final List<IToken<?>> tokens = new ArrayList<>();
+        for (final Item item : Tags.Items.SAND.getAllElements())
+        {
+            final ItemStack stack = new ItemStack(item);
+            final ItemStack output = FurnaceRecipes.getInstance().getSmeltingResult(stack);
+            if (Tags.Items.GLASS.contains(output.getItem()))
+            {
+                final List<ItemStack> list = new ArrayList<>();
+                list.add(stack);
+
+                final IRecipeStorage storage = StandardFactoryController.getInstance().getNewInstance(
+                  TypeConstants.RECIPE,
+                  StandardFactoryController.getInstance().getNewInstance(TypeConstants.ITOKEN),
+                  list,
+                  1,
+                  output,
+                  Blocks.FURNACE);
+                final IToken<?> token = IColonyManager.getInstance().getRecipeManager().checkOrAddRecipe(storage);
+                if (recipes.contains(token))
+                {
+                    return;
+                }
+                tokens.add(token);
+            }
+        }
+        recipes.addAll(tokens);
```

Comment:
This recipes.addAll looks like it could cause duplication, as it doesn't have the protection of addRecipeToList.

Reply:
Check in the loop before, if any of that two is in the list, it returns